### PR TITLE
feat: Balancer v2 to studio

### PR DESCRIPTION
**Pull request by wpoulin**
## Description

Migrated balancer-v2

## Checklist

- [X] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)

## How to test?

arb - `0x0020c5222a24e4a96b720c06b803fb8d34adc0af`
eth - `0x0020c5222a24e4a96b720c06b803fb8d34adc0af`
poly - `0x53f2736fae551c998d4e72e519f1acf474264de4`

### CODE REVIEW

File: src/apps/balancer-v2/balancer-v2.module.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { BalancerV2ClaimableContractPositionBalanceHelper } from './helpers/balancer-v2.claimable.contract-position-balance-helper';
+import { BalancerV2EventsPoolTokenDataStrategy } from './helpers/balancer-v2.events.pool-token-address-strategy';
+import { BalancerV2GaugeAddressesGetter } from './helpers/balancer-v2.gauge-addresses-getter';
+import { BalancerV2PoolTokensHelper } from './helpers/balancer-v2.pool.token-helper';
+import { BalancerV2SpotPriceHelper } from './helpers/balancer-v2.spot-price.helper';
+import { BalancerV2TheGraphPoolTokenDataStrategy } from './helpers/balancer-v2.the-graph.pool-token-address-strategy';
+import { PolygonBalancerV2BalanceFetcher } from './polygon/balancer-v2.balance-fetcher';
+import { PolygonBalancerV2PoolTokenFetcher } from './polygon/balancer-v2.pool.token-fetcher';
+
+@Module({
```
Comment (immasandwich) [inline src/apps/balancer-v2/balancer-v2.module.ts:24]: Use `@Register.AppModule`

File: src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// What in the cinnamon toast fuck is this? Gather around children...
+// 1. Balancer maintains off-chain claimable amount snapshots in the JSON files described by the `manifestUrl` property
+// 2. Originally, Balancer used the `legacyMerkleRedeemAddress` to maintain claim states
+// 3. Now, Balancer uses the `merkleOrchardAddress` to maintain claim states (after `weekStart`)
+// 4. Gather the states from both to retrieve the full claimable amount state.
+
+// Resources:
+// Legacy merkle redeem configurations can be found in https://github.com/balancer-labs/erc20-redeemable
+// New merkle orchard configurations can be found in https://github.com/balancer-labs/frontend-v2
```
Comment (immasandwich) [inline src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts:14]: These comments can be deleted, and possibly the claimable config file as well.

File: src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+@Injectable()
+export class BalancerV2CacheManager {
+  constructor() {}
+
+  async getCachedClaimableAmounts(opts: Opts): Promise<number[]> {
+    const { address, network, rewardTokenAddress } = opts;
+
+    const url = `https://us-west1-zapper-backend.cloudfunctions.net/balancer-v2-claimable`;
+    const response = await Axios.get(url, { params: { address, network, rewardTokenAddress } });
```
Comment (immasandwich) [inline src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts:41]: Type this please :)

File: src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+// 1. Balancer maintains off-chain claimable amount snapshots in the JSON files described by the `manifestUrl` property
+// 2. Originally, Balancer used the `legacyMerkleRedeemAddress` to maintain claim states
+// 3. Now, Balancer uses the `merkleOrchardAddress` to maintain claim states (after `weekStart`)
+// 4. Gather the states from both to retrieve the full claimable amount state.
+
+// Resources:
+// Legacy merkle redeem configurations can be found in https://github.com/balancer-labs/erc20-redeemable
+// New merkle orchard configurations can be found in https://github.com/balancer-labs/frontend-v2
+
+export type ClaimableTokenConfig = {
```
Comment (immasandwich) [inline src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts:16]: Delete unused type

File: src/types/constant.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Network } from './network.interface';
+
+export const chainIdByNetwork: Partial<Record<Network, number>> = {
```
Comment (immasandwich) [inline src/types/constant.ts:3]: Use `src/types/network.interface.ts`

File: src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts (hunk 4 above)
Comment (wpoulin) [reply src/apps/balancer-v2/helpers/balancer-v2.cache-manager.ts:16]: I'll move it next to it's utilization in `balancer-v2.claimables.config.ts`

**Review by immasandwich — APPROVED** [review]

**Pull request merged**